### ISSUE TITLE
Crash on Top Site long press share action

### ISSUE THREAD

**Issue by lmarceau** [mozilla-mobile/firefox-ios#9806]
Discovered while investigating issue https://github.com/mozilla-mobile/firefox-ios/issues/9361

## Steps to reproduce
1. On iPad - happens on latest release version too (v96)
2. Long press on a top site shortcut
3. Tap "Share"

### Expected behavior
Doesn't crash and opens the share menu

### Actual behavior
Crashes the app with following stack trace:
```
2022-01-18 19:35:02.924532-0500 Client[59904:665747] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'UIPopoverPresentationController (<UIPopoverPresentationController: 0x7f8295d329d0>) should have a non-nil sourceView or barButtonItem set before the presentation occurs.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e3de6e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff512a19b2 objc_exception_throw + 48
	2   UIKitCore                           0x00007fff48bcfd0a __66-[UIPopoverPresentationController presentationTransitionWillBegin]_block_invoke + 0
	3   UIKitCore                           0x00007fff48bda9e8 __71-[UIPresentationController _initViewHierarchyForPresentationSuperview:]_block_invoke + 2646
	4   UIKitCore                           0x00007fff48bd823f __56-[UIPresentationController runTransitionForCurrentState]_block_invoke.465 + 546
	5   UIKitCore                           0x00007fff4937bb72 _runAfterCACommitDeferredBlocks + 352
	6   UIKitCore                           0x00007fff4936c300 _cleanUpAfterCAFlushAndRunDeferredBlocks + 248
	7   UIKitCore                           0x00007fff4939cb09 _afterCACommitHandler + 85
	8   CoreFoundation                      0x00007fff23da1087 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	9   CoreFoundation                      0x00007fff23d9bb3e __CFRunLoopDoObservers + 430
	10  CoreFoundation                      0x00007fff23d9c08a __CFRunLoopRun + 1226
	11  CoreFoundation                      0x00007fff23d9b8a4 CFRunLoopRunSpecific + 404
	12  GraphicsServices                    0x00007fff38c05bbe GSEventRunModal + 139
	13  UIKitCore                           0x00007fff49372964 UIApplicationMain + 1605
	14  Client                              0x0000000105912277 main + 455
	15  ???                                 0x000000010dc7d4fe 0x0 + 4526167294
)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSGenericException', reason: 'UIPopoverPresentationController (<UIPopoverPresentationController: 0x7f8295d329d0>) should have a non-nil sourceView or barButtonItem set before the presentation occurs.'
```

### Device & build information
* Device: iPad mini 6th gen (15.1) && iPad Pro 13.7
* Build version: Any

### Notes
Attachments:

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3640)

**Comment by lmarceau** [comment]
Note: The same crash was also happening on iPad on the long press for the Pocket cells (with the share action). Was also fixed in the same PR.

**Comment by data-sync-user** [comment]
➤ Simion Basca commented:

Verifying as fix on main c6d17c95da56477529f3bc3a1bd0dbf50d2717a5.
